blog/2024-04-30-nushell_0_93_0.md

```diff
@@ -80,6 +80,91 @@ Thanks to all the contributors below for helping us making the documentation of
 | ------------------------------------ | ----------- | ------------------------------------------------------- |
 | [@author](https://github.com/author) | ...         | [#12345](https://github.com/nushell/nushell/pull/12345) |
 
+## A new crate: `nuon`
+NUON is the NUshell Object Notation, a superset of JSON that allows to store any kind of data.
```

Comment:
"store any kind of data" seems like an overstatement because there are several nushell data types that nuon cannot represent already. Maybe I'm just stuck in the details?

Reply:
closures are not really data, right? that's the reason why they can't be serialized to NUON afaik

are there other types of data that can't be converted to NUON that i'm not thinking of? :blush:

Reply:
From looking at to.rs in nuon, these are the nushell Values that cannot be converted to nuon.
- `Closure`
- `CellPath`
- `Custom`
- `Error`

As far as closures are concerned, I'm not sure why they couldn't be represented as strings like `{|| do something}`. Also, cell-paths could be `$.whatever.the.path` maybe. We just have to be able to use `from nuon` in them if we did that.

Reply:
yeah, i share the same feeling, cell paths should be possible to serialize to NUON and some closures, probably the ones that don't capture any outer scope, too :yum:

Reply:
Ok, so that was my point, that it can't serialize `any` kind of data. That's all.